### DISCUSSION TITLE
Restrict File Access to USB Devices Only on Linux

### DISCUSSION THREAD

**Discussion by IdoPractical** [tauri-apps/tauri#10502] [Q&A] [answered]
Hi,

I'm working on an application where I need to restrict users to open files only from USB storage devices on a Linux system. I understand that setting the scope for the fs module in tauri.conf.json can limit file access paths, but this doesn't prevent the open file dialog from showing other locations on the system.

Is there a way to configure the Tauri application so that the open file dialog only displays USB storage devices and prevents access to any other directories? Ideally, the user should only be able to see and select files from USB devices mounted under typical mount points like /media or /mnt.

Any guidance or suggestions on how to achieve this would be greatly appreciated.

Thank you!

**Comment by FabianLars** [comment]
I don't think the os/gtk file picker APIs we use have any way of doing this but if you ship your app as a flatpak you may be able to leverage its sandbox for that 🤔

**Reply by IdoPractical** [reply]
Thank you!
What if I want to restrict it with the FS on Tauri?
Then I can pop up a notification to the user saying it's not a valid path.
I tried setting the FS in `tauri.conf.json` like this:
```json
      "fs": {
        "all": false,
        "copyFile": false,
        "createDir": false,
        "exists": false,
        "readDir": false,
        "readFile": true,
        "removeDir": false,
        "removeFile": false,
        "renameFile": false,
        "scope": ["/mnt/*"],
        "writeFile": true
      }
```
but it didn't work and the program accepted files from the /home folder.

**Reply by FabianLars** [reply]
Just so we're talking about the same thing: The fs scope won't affect the dialog.

That said, it should restrict the read/write permissions to just /mnt (your scope is correct), _but_ tauri's dialog api will extend the scope at runtime for the selected paths... So then you'd have to check the paths yourself (preferebly in rust because then you can modify the scope again).

**Reply by IdoPractical** [reply]
I know it won't affect the dialog.
Just let me get the point, so you are saying that the restriction on the json file will be overridden by the dialog api thus I need to check the paths on rust.
Is that correct?
Do you have an example for changing the scope via rust?

**Reply by FabianLars** [reply, marked as answer]
> Just let me get the point, so you are saying that the restriction on the json file will be overridden by the dialog api thus I need to check the paths on rust.
Is that correct?

Yes.

>  Do you have an example for changing the scope via rust?

Hmm, i don't think i have anything with context, but essentially once you have access to an AppHandle (for example in [commands](https://tauri.app/v1/guides/features/command/#accessing-an-apphandle-in-commands)) you essentially do `app_handle.fs_scope().allow_file(path)` or any of these apis: https://docs.rs/tauri/latest/tauri/scope/struct.FsScope.html (should be the same for v2)

**Reply by IdoPractical** [reply]
Thank you for your help!
